src/mutable_data.rs

```diff
@@ -7,58 +7,117 @@
+/// Mutable data that is unpublished on the network. This data can only be fetch be the owners / those in the permissions fields with `Permission::Read` access.
 #[derive(Hash, Eq, PartialEq, PartialOrd, Ord, Clone, Serialize, Deserialize)]
-pub enum MutableDataKind {
-    // Unsequenced, unpublished Mutable Data.
-    Unsequenced { data: BTreeMap<Vec<u8>, Value> },
-    // Sequenced, unpublished Mutable Data.
-    Sequenced { data: BTreeMap<Vec<u8>, Vec<u8>> },
+pub struct SeqMutableData {
+    /// Network address.
+    name: XorName,
+    /// Type tag.
+    tag: u64,
+    /// Key-Value semantics.
+    data: BTreeMap<Vec<u8>, Value>,
+    /// Maps an application key to a list of allowed or forbidden actions.
+    permissions: BTreeMap<PublicKey, PermissionSet>,
+    /// Version should be increased for any changes to MutableData fields except for data.
+    version: u64,
+    /// Contains a set of owners of this data. DataManagers enforce that a mutation request is
+    /// coming from the MaidManager Authority of the Owner.
+    owners: PublicKey,
 }
 
-/// Permissions given to a user / application for a mutable data field.
-#[derive(Hash, Eq, PartialEq, PartialOrd, Ord, Clone, Serialize, Deserialize)]
-pub enum Permission {
-    Read,
-    Insert,
-    Update,
-    Delete,
-    ManagePermissions,
+/// A value in `Sequenced MutableData`
+#[derive(Hash, Eq, PartialEq, PartialOrd, Ord, Clone, Serialize, Deserialize, Debug)]
+pub struct Value {
+    /// Actual data.
+    data: Vec<u8>,
+    /// SHALL be incremented sequentially for any change to `data`.
+    version: u64,
 }
 
-/// Mutable data that is unpublished on the network. This data can only be fetch be the owners / those in the permissions fiedls with `Permission::Read` access.
+/// Mutable data that is unpublished on the network. This data can only be fetch be the owners / those in the permissions fields with `Permission::Read` access.
 #[derive(Hash, Eq, PartialEq, PartialOrd, Ord, Clone, Serialize, Deserialize)]
-pub struct MutableData {
+pub struct UnseqMutableData {
     /// Network address.
     name: XorName,
     /// Type tag.
     tag: u64,
     /// Key-Value semantics.
-    data: MutableDataKind,
+    data: BTreeMap<Vec<u8>, Vec<u8>>,
     /// Maps an application key to a list of allowed or forbidden actions.
-    permissions: BTreeMap<User, BTreeSet<Permission>>,
+    permissions: BTreeMap<PublicKey, PermissionSet>,
     /// Version should be increased for any changes to MutableData fields except for data.
     version: u64,
     /// Contains a set of owners of this data. DataManagers enforce that a mutation request is
     /// coming from the MaidManager Authority of the Owner.
-    /// Currently limited to one owner to disallow multisig.
     owners: PublicKey,
 }
 
-impl MutableData {
+/// Set of user permissions.
+#[derive(Clone, Debug, Eq, Hash, PartialEq, PartialOrd, Ord, Serialize, Deserialize, Default)]
+pub struct PermissionSet {
+    permissions: BTreeSet<Action>,
```

Comment:
Eventually it'd be better to replace `PermissionSet` with just `BTreeSet<Action>`, but now for compatibility it's a good idea to leave it as it is. Nice one :+1: